pkg/descheduler/client/client.go

```diff
@@ -20,38 +20,46 @@ import (
 	"fmt"
 
 	clientset "k8s.io/client-go/kubernetes"
+	componentbaseconfig "k8s.io/component-base/config"
+
 	// Ensure to load all auth plugins.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func CreateClient(kubeconfig, userAgt string) (clientset.Interface, error) {
+func CreateClient(clientConnection componentbaseconfig.ClientConnectionConfiguration, userAgt string) (clientset.Interface, error) {
 	var cfg *rest.Config
-	if len(kubeconfig) != 0 {
-		master, err := GetMasterFromKubeconfig(kubeconfig)
+	if len(clientConnection.Kubeconfig) != 0 {
+		master, err := GetMasterFromKubeconfig(clientConnection.Kubeconfig)
 		if err != nil {
-			return nil, fmt.Errorf("Failed to parse kubeconfig file: %v ", err)
+			return nil, fmt.Errorf("failed to parse kubeconfig file: %v ", err)
 		}
 
-		cfg, err = clientcmd.BuildConfigFromFlags(master, kubeconfig)
+		cfg, err = clientcmd.BuildConfigFromFlags(master, clientConnection.Kubeconfig)
 		if err != nil {
-			return nil, fmt.Errorf("Unable to build config: %v", err)
+			return nil, fmt.Errorf("unable to build config: %v", err)
 		}
 
 	} else {
 		var err error
 		cfg, err = rest.InClusterConfig()
 		if err != nil {
-			return nil, fmt.Errorf("Unable to build in cluster config: %v", err)
+			return nil, fmt.Errorf("unable to build in cluster config: %v", err)
 		}
 	}
 
+	if clientConnection.Burst != 0 {
+		cfg.Burst = int(clientConnection.Burst)
+	}
+	if clientConnection.QPS != 0 {
+		cfg.QPS = clientConnection.QPS
+	}
```

Comment:
is there any meaning to the user explicitly setting `0` for these?

Reply:
I don't think so, since that's the default value. In fact, I should remove the conditionals